server/supernova/sc/sc_osc_handler.cpp

```diff
@@ -1480,7 +1480,16 @@ void g_query_tree_fill_node(osc::OutboundPacketStream & p, bool flag, server_nod
             p << controls;
 
             for (int i = 0; i != controls; ++i) {
-                p << osc::int32(i); /** \todo later we can return symbols */
+                const char * name_of_slot;
+                if( i < scsynth.number_of_slots())
+                    name_of_slot = scsynth.name_of_slot(i);
+                else
+                    name_of_slot = 0;
+
+                if(name_of_slot)
+                    p << name_of_slot;
+                else
+                    p << osc::int32(i);
```

Comment:
This can be simplified to a single branch:

```cpp
if (i < scsynth.number_of_slots())
  p << scsynth.name_of_slot(i);
else
  p << osc::int32(i);
```

also style note - put a space after `if` and not after the open paren